### PR TITLE
WIP: Integrates Google Login API in the application

### PR DESCRIPTION
**Pull request by JCruz-Dev**
Hello!

this Pull Request Integrates the Google API for the Sign in and Sign Out saving the token given by google on the LocalStorage for now. This will be a WIP PR as @GianCastle  told me.

Things that are still in progress:
- Render login page at first because right now when you start the local server, it takes you to the jobs page
- after succesful the login, redirect the user to the dashboard page
- authenticate every route

### CODE REVIEW

File: .default.env [hunk 1]
```diff
@@ -1,5 +0,0 @@
-EXTEND_ESLINT=true
-REACT_APP_ENVIROMENT=stage
-REACT_APP_APPTITLE=ADMIN.EMPLEA.DO
-REACT_APP_PROXY_URL=https://cors-anywhere.herokuapp.com
-REACT_APP_API_URL=https://api-emplea-do.herokuapp.com
```
Comment (GianCastle) [inline .default.env:5]: This file served as an example to configure a .env in development environments. Why did you decide to remove it? 😨

File: src/components/Login/SignIn.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React, { useEffect } from 'react'
+import { connect } from 'react-redux'
+import { useLogin } from 'react-admin'
+
+import AuthProvider from '../../providers/authProvider'
+
+const SignIn = props => {
+  function onSignIn() {
```
Comment (GianCastle) [inline src/components/Login/SignIn.jsx:8]: Just for convetions, you should use const whereever you can use it.

File: src/components/Login/SignIn.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    AuthProvider.login()
+    if (AuthProvider.checkAuth()) {
```
Comment (GianCastle) [inline src/components/Login/SignIn.jsx:10]: Please, avoid empty `if` clauses

File: .default.env (hunk 1 above)
Comment (JCruz-Dev) [reply .default.env:5]: All the constants were  giving undefined and just renamed the file to `.env` but if that file it wasnt supposed to be deleted I apologize

File: src/components/Login/SignIn.jsx (hunk 2 above)
Comment (JCruz-Dev) [reply src/components/Login/SignIn.jsx:8]: Ohh ok. Thanks for the feedback. I will

File: src/index.js [hunk 4]
```diff
@@ -3,6 +3,9 @@ import ReactDOM from 'react-dom'
 
 import App from './containers/App'
 import * as serviceWorker from './serviceWorker'
+import AuthProvider from './providers/authProvider'
+
+AuthProvider.initOauth()
```
Comment (GianCastle) [inline src/index.js:8]: This function is called on this file, which means on every screen we call  the`iniOAuth` method and this method is called over and over and over

File: src/providers/authProvider.js [hunk 5]
```diff
@@ -3,30 +3,47 @@
...
-    return Promise.resolve()
+  checkError: googleUser => {
+    console.log(googleUser)
   },
   // called when the user navigates to a new location, to check for authentication
   checkAuth: () => {
-    return localStorage.getItem('username')
-      ? Promise.resolve()
-      : Promise.reject()
+    return localStorage.getItem('user') ? Promise.resolve() : Promise.reject()
```
Comment (GianCastle) [inline src/providers/authProvider.js:46]: Resolve linting and code styling conflicts.

File: public/index.html [hunk 6]
```diff
@@ -39,5 +39,6 @@
       To begin the development, run `npm start` or `yarn start`.
       To create a production bundle, use `npm run build` or `yarn build`.
     -->
+    <script src="https://apis.google.com/js/platform.js" async defer></script>
```
Comment (GianCastle) [inline public/index.html:42]: Avoid to call external scripts from React

**Review by GianCastle — CHANGES_REQUESTED** [review]
For this flow of authentication the following issues arise

- If I authenticate with any user, in practice any user could access the administrative resources of the portal
- If I authenticate as any user, I would not have a token with administrative roles that would allow me to request resources from the API
- This OAuth flow is entirely dependent on Google Auth Platofrm by loading an external script. In case another authentication strategy is required, this flow would not be flexible enough to be extended.

**Pull request closed without merging**